src/charm.py

```diff
@@ -1,427 +1,127 @@
 #!/usr/bin/env python3
-# Copyright 2020 Luke Marsden
+# Copyright 2021 Canonical Ltd.
 # See LICENSE file for licensing details.
+#
 
-"""Charm for the ML Flow Server.
-
-https://github.com/canonical/mlflow-operator
-"""
-
-import json
 import logging
-from base64 import b64encode
 
-from charms.grafana_k8s.v0.grafana_dashboard import GrafanaDashboardProvider
-from charms.prometheus_k8s.v0.prometheus_scrape import MetricsEndpointProvider
-from oci_image import OCIImageResource, OCIImageResourceError
+from charmed_kubeflow_chisme.exceptions import ErrorWithStatus
+from charms.observability_libs.v1.kubernetes_service_patch import KubernetesServicePatch
+from lightkube.models.core_v1 import ServicePort
 from ops.charm import CharmBase
 from ops.main import main
-from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, StatusBase, WaitingStatus
-from serialized_data_interface import NoCompatibleVersions, NoVersionsListed, get_interfaces
-
-from services.s3 import S3BucketWrapper, validate_s3_bucket_name
-
-PROMETHEUS_PATH = "/metrics"
-
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
+from ops.pebble import ChangeError, Layer
 
-class Operator(CharmBase):
-    """Charm for the ML Flow Server.
 
-    https://github.com/canonical/mlflow-operator
-    """
+class MlflowCharm(CharmBase):
+    """A Juju Charm for Training Operator"""
```

Comment:
```suggestion
    """A Juju Charm for MLFlow."""
```

Reply:
Can you check your linting config vs our other charms?  I'd have expected this to fail because it doesn't end in ".", so I think the config is not the same as our other charms